plugins/hls-refactor-plugin/src/Development/IDE/Plugin/Plugins/FillHole.hs

```diff
@@ -20,18 +20,24 @@ suggestFillHole Diagnostic{_range=_range,..}
         ++ map (proposeHoleFit holeName True isInfixHole) refFits
     | otherwise = []
     where
-      extractHoleName = fmap head . flip matchRegexUnifySpaces "Found hole: ([^ ]*)"
+      extractHoleName = fmap (headOrThrow "impossible") . flip matchRegexUnifySpaces "Found hole: ([^ ]*)"
       addBackticks text = "`" <> text <> "`"
       addParens text = "(" <> text <> ")"
       proposeHoleFit holeName parenthise isInfixHole name =
-        let isInfixOperator = T.head name == '('
-            name' = getOperatorNotation isInfixHole isInfixOperator name in
-          ( "replace " <> holeName <> " with " <> name
-          , TextEdit _range (if parenthise then addParens name' else name')
-          )
+        case T.uncons name of
+          Nothing -> error "impossible: empty name provided by ghc"
```

Comment:
To be clear, these are an improvement in that they give clearer errors, but they will still _crash HLS unrecoverably_. We really, really want to avoid `error` and friends also. But fixing them properly would require more refactoring.